src/libecalc/presentation/yaml/domain/components/electricity_consumer_component.py

```diff
@@ -0,0 +1,95 @@
+from libecalc.core.result import EcalcModelResult
+from libecalc.core.result.emission import EmissionResult
+from libecalc.dto import ElectricEnergyUsageModel, ElectricityConsumer
+from libecalc.dto.component_graph import PowerConsumer
+from libecalc.dto.node_info import NodeInfo
+from libecalc.expression import Expression
+from libecalc.presentation.yaml.domain.reference_service import ReferenceService
+from libecalc.presentation.yaml.mappers.component_mapper import ConsumerMapper
+from libecalc.presentation.yaml.yaml_types.components.legacy.yaml_electricity_consumer import YamlElectricityConsumer
+
+
+class ElectricityConsumerComponent(PowerConsumer):
+    def __init__(
+        self,
+        yaml_fuel_consumer: YamlElectricityConsumer,
+        reference_service: ReferenceService,
+        target_period: Period,
+        regularity: Dict[datetime, Expression],
+        default_fuel_reference: Optional[str],
+        expression_evaluator: ExpressionEvaluator,
+    ):
+        self._expression_evaluator = expression_evaluator
+        self._yaml_consumer = yaml_fuel_consumer
+        consumer_mapper = ConsumerMapper(references=reference_service, target_period=target_period)
+        self._dto_consumer: ElectricityConsumer = consumer_mapper.from_yaml_to_dto(
+            yaml_fuel_consumer,
+            regularity=regularity,
+            consumes=ConsumptionType.ELECTRICITY,
+            default_fuel=default_fuel_reference,
+        )
+        self._core_consumer = Consumer(
+            id=self.id,
+            name=self.name,
+            component_type=self.get_node_info().component_type,
+            regularity=TemporalModel(self._dto_consumer.regularity),
+            consumes=ConsumptionType.ELECTRICITY,
+            energy_usage_model=TemporalModel(
+                {
+                    start_time: EnergyModelMapper.from_dto_to_domain(model)
+                    for start_time, model in self._dto_consumer.energy_usage_model.items()
+                }
+            ),
+        )
+
+        # Keep results when calculated
+        self._ecalc_model_result: Optional[EcalcModelResult] = None
+        self._emissions: Optional[Dict[str, EmissionResult]] = None
+
+    @property
+    def id(self) -> NodeID:
+        return generate_id(self._yaml_consumer.name)
+
+    @property
+    def name(self) -> str:
+        return self._yaml_consumer.name
+
+    def get_node_info(self) -> NodeInfo:
+        return NodeInfo(
+            id=self.id,
+            name=self.name,
+            component_level=ComponentLevel.CONSUMER,
+            component_type=self._dto_consumer.component_type,
+        )
+
+    def get_power_requirement(self, period: Period = None) -> TimeSeriesStreamDayRate:
+        consumer_result = self._core_consumer.evaluate(self._expression_evaluator)
+        self._ecalc_model_result = consumer_result
+        return consumer_result.component_result.power
+
+    def get_ecalc_model_result(self) -> EcalcModelResult:
```

Comment:
I also think we should also aim to remove EcalcModelResult, instead provide specific results such as Streams, fuel usage, power usage, emissions. Per equipment results should also be considered. Instead of having EcalcModelResult for FuelConsumer that contains both fuel and power usage, we could have two sets of results, one that represents the turbine, and another that represents the compressor. Turbine naturally has fuel usage, compressor has power usage. If the fuel consumer doesn't provide power usage, there's only a turbine result.

We should think about this when defining the different types of components we have. each component might contain several components in the `result` context, but not in the evaluation context (Similar to ProcessGraph/ConsumerSystem).